app/role.go

```diff
@@ -20,11 +20,62 @@ func (a *App) GetAllRoles() ([]*model.Role, *model.AppError) {
 }
 
 func (a *App) GetRoleByName(name string) (*model.Role, *model.AppError) {
-	return a.Srv().Store.Role().GetByName(name)
+	role, err := a.Srv().Store.Role().GetByName(name)
+	if err != nil {
+		return nil, err
+	}
+
+	err = a.updateChannelRolesPermissions([]*model.Role{role})
```

Comment:
I feel like this method could be named differently - it implies that the `Role` is being updated i.e. changes are being committed to the database but thats not actually the case its simply merging the permissions and returning the merged permissions right?

Reply:
Yes you're correct about the behaviour: it's not persisting the change to the DB. Thoughts about a better name?

Reply:
@fm2munsh I wonder if this should be a function that returns the merged permissions rather than a method on `Role` that updates the permissions in-place? That's what I had originally, actually.

Reply:
yea i think it would be better if it returned the merged permissions instead of modifying the actual struct 👍

Reply:
maybe it could be `getMergedChannelPermissions`?

Reply:
Hmm... actually I was thinking of `(*Role).MergeHigherScopedPermissions`, actually, when I suggested returned the changed permissions rather than replacing them.

`(*App).updateChannelRolesPermissions` is used in 3 places and all of them really need to replace the permissions on the role structs, and because it operates on a slice of roles it wouldn't make sense to not replace them on the role structs themselves.

Perhaps we can think of a name that doesn't imply that it's persisted but still allows them method to mutate the roles like `mutateChannelRolesPermissions`?

Reply:
I think it could be called `(*App).mergeInheritedChannelPermissions` - Also might be worth it to rename instances `HigherScoped` to `Inherited` unless you prefer the wording `HigherScoped`?

Reply:
and `LowerScoped` can be `Inheriting`

Reply:
I think we should keep the "higher-scope" terminology for now. At Mattercon I know we're going to discuss changing the architecture to make inheritance of permissions more standard across all scopes (not just for channels), etc.. I think deciding on the terminology based on those decisions probably makes sense.